AttributeError: module 'prometheus_client.registry' has no attribute 'Collector' on el8

**Describe the bug**
```
>   class SlurmWebMetricsCollector(prometheus_client.registry.Collector):
E   AttributeError: module 'prometheus_client.registry' has no attribute 'Collector'
```

**To Reproduce**
Enable metrics feature on el8 or run tests with prometheus_client available in el8.

**Expected behavior**
This exception must not be raised.

**Versions**
v4

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [unreleased]
+
+### Fixed
+- Fix `AttributeError` with `prometheus_client.registry.Collector` on el8 (#548)
+
 ## [4.2.0] - 2025-04-29
 
 ### Added

--- a/slurmweb/metrics/collector.py
+++ b/slurmweb/metrics/collector.py
@@ -11,6 +11,13 @@ import logging
 import prometheus_client
 import prometheus_client.core
 
+# In Prometheus client < v0.14.0 (distributed in EPEL8), Collector abstract
+# class is not defined. Use concrete CollectorRegistry class as an alternative.
+try:
+    from prometheus_client.registry import Collector
+except ImportError:
+    from prometheus_client.registry import CollectorRegistry as Collector
+
 from ..errors import SlurmwebCacheError
 from ..slurmrestd.errors import (
     SlurmrestdNotFoundError,
@@ -25,7 +32,7 @@ if t.TYPE_CHECKING:
 logger = logging.getLogger(__name__)
 
 
-class SlurmWebMetricsCollector(prometheus_client.registry.Collector):
+class SlurmWebMetricsCollector(Collector):
     def __init__(self, slurmrestd):
         self.slurmrestd = slurmrestd
         self.register()

